test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/schema/SchemaComplianceChecker.java

```diff
@@ -0,0 +1,371 @@
+package uk.gov.justice.services.test.utils.core.schema;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.counting;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+
+import difflib.Delta;
+import difflib.DiffUtils;
+import difflib.Patch;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SchemaComplianceChecker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SchemaComplianceChecker.class);
+
+    private static final String WINDOWS_NEWLINE = "\r\n";
+    private static final String UNIX_NEWLINE = "\n";
+
+    private static final String JSON_SCHEMA_FILE_EXTENSION = ".json";
+    private static final String JSON_SCHEMA_BASE_PATH = "json/schema/";
+
+    private int totalSchemasChecked = 0;
+    private boolean isCompliant = false;
+    private int numberOfViolatingSchemas = 0;
+
+    public boolean isCompliant() {
+        return isCompliant;
+    }
+
+
+    /**
+     * Runs checks to identify schemas with the same filename but with different content across
+     * modules
+     */
+    public void runComplianceCheck() throws SchemaComplianceException {
+        this.isCompliant = false;
+        this.totalSchemasChecked = 0;
+        this.numberOfViolatingSchemas = -1;
```

Comment:
Why do we need to -1 this?

Reply:
Magic number which would allow me to understand from outside the class if the code had been executed to the very end. will make this 0 and change the return type to boolean